### ISSUE TITLE
I can't seem to make an external MPV player honor its own mpv.conf and input.conf 

### ISSUE THREAD

**Issue by movieperson** [jellyfin/jellyfin-mpv-shim#341]
I am attempting to use a heavily customized version of MPV and have it only use its own configuration file as it does when it is directly invoked from the command line.  (start /min D:\apps\mpv_default\mpv.exe). 

From my conf.json

    "mpv_ext": true,
    "mpv_ext_ipc": "MyIPCPath" , (Note: I tried leaving this blank as well)  When using this parameter, I tried starting mpv with
                                                                                      start /min D:\apps\mpv_default\mpv.exe --input-ipc-server=MyIPCPath          
    "mpv_ext_no_ovr": true,  (Note I tried both true and false for this parameter)
    "mpv_ext_path": "D:\\Apps\\mpv_default\\mpv.exe",
    "mpv_ext_start": false,  (Note I tried both true and false for this parameter)
    "mpv_log_level": "info",

Is it me, or is another issue preventing this from working?  Or is it working as designed?

my modified MPV config files are attached.

Thanks in advance for any help as I wasn't exactly sure how to classify it.

[input.conf.txt](https://github.com/jellyfin/jellyfin-mpv-shim/files/11738763/input.conf.txt)
[mpv.conf.txt](https://github.com/jellyfin/jellyfin-mpv-shim/files/11738766/mpv.conf.txt)

**Comment by dmgolembiowski** [comment]
@movieperson are you comfortable tweaking some Python on your local machine? I don't have a Windows environment to test this on, but I think the breakdown is occuring at a few key locations, however I've included a sanity check below which should be done before inspecting the bug.

#### Sanity Check
@movieperson If I correctly understand how your desired setup is supposed to work, let me verify a couple of assumptions. Assuming you are running `jellyfin-mpv-shim.exe` from CMD, it will have a `%PATH%` variable. Try running `echo %PATH%` for a priority-ordered list of folders that hold your binaries. You want to check for a few things. First, please check if there is more than one installation of either `jellyfin-mpv-shim.exe` or `mpv.exe`, and verify that their versions correspond to the build you want. If both of these are not present on `%PATH%`, you might want to extend your [environment variable in your system settings](https://docs.oracle.com/en/database/oracle/machine-learning/oml4r/1.5.1/oread/creating-and-modifying-environment-variables-on-windows.html#GUID-DD6F9982-60D5-48F6-8270-A27EC53807D0). It's worth noting that you need to open a new CMD terminal to have these take effect, or simply log out and log back in to have the fresh values loaded into the environment.

The next thing to consider is that `"mpv_ext_ipc"` corresponds to the [executable file's absolute path](https://github.com/jellyfin/jellyfin-mpv-shim/blob/7b9b7c9230d7bfcc3cb0ead560de4e213704f848/README.md?plain=1#L254). For your specific case, the key-value pair in `conf.json` should be:

```json
"mpv_ext": true,
"mpv_ext_ipc": "D:\\apps\\mpv_default\\mpv.exe",
```

If your computer runs multiple different versions of Python, check that `%PATH%` variable and verify it's the one corresponding to the build version you need.

#### Possible source of bug
This last bit is probably where one of your problems is occurring, and it's a subtle thing specific to the MPV player this attempt tries to construct. On Windows, the system call to remove the IPC socket fails because it is still being held by an external process (your `start /min ...` command). Because of this, the process terminates the IPC socket that would have been created by the jellyfin-mpv-shim (for @iwalton3 see https://github.com/iwalton3/python-mpv-jsonipc/blob/master/python_mpv_jsonipc.py#L248). For @movieperson this means you do not want to have an existing MPV process running elsewhere when you want to use your customized setup. The child process needs to be spawned by this application, otherwise it will fall back on trying to spin up an instance that works.
This seems to directly contradict the third bullet point of [README.md](https://github.com/jellyfin/jellyfin-mpv-shim/tree/7b9b7c9230d7bfcc3cb0ead560de4e213704f848#external-mpv) on Windows.

**Comment by movieperson** [comment]
Note that I am running mpv in portable_config mode, so all the config files are in that subdirectory.

What's not clear to me is whether there should even be an mpv.conf and an input.conf file in the portable_config directory or if the files with that name in the roaming\jellyfin directory take their place or override them.

I did create a batch file and added the paths.  
set PATH=%PATH%;D:\apps\mpv_default;C:\Program Files (x86) 
"C:\Program Files (x86)\Jellyfin MPV Shim\run.exe"

**Comment by movieperson** [comment]
Making some progress, although looking in task manager I notice I have a bunch of run.exe and runbroker tasks left over when I end MPV.

I think the issue I am having is mostly conflicts between my key defs and the defaults.  Stuff that works in native MPV doesn't do the same function, or more likely just hangs.  I have deleted the mpv.conf and input.conf from the MPV portable config directory.

It would help if you could publish base working input.conf and mpv.conf files.  for example, I have not been able to bring up the jellyfin menu with the "c" key, although I vaguely remember that working when I first installed Jellyfin_shim.  Simply copying my MPV conf files just doesn't work, even though they work perfectly in native MPV.

Thanks

**Comment by soredake** [comment]
Same problem.
